lib/fs.ts

```diff
@@ -164,7 +164,7 @@ class FSPath {
 			return;
 		}
 
-		this.writeUpdateNow(dataFetcher, options);
+		return this.writeUpdateNow(dataFetcher, options);
```

Comment:
This one shouldn't be returned, though.

The old implementation of this function was actually async, and the doc comment said "don't use the returned Promise for anything; it's meaningless".

After this function was refactored into `writeUpdateNow`/`checkNextUpdate`/`finishUpdate`, it intentionally no longer returns anything. This return value would in fact still be meaningless.

(`writeUpdateNow`'s return value is irrelevant, and I don't have a good sense for whether or not `checkNextUpdate` should return a `Promise`.)

Reply:
```suggestion
		// tslint:disable-next-line:no-floating-promises
		this.writeUpdateNow(dataFetcher, options);
```

Reply:
As a recap: `writeUpdate` will write an update the next time it's safe to do so.

If a previous call to `writeUpdate` hasn't finished writing, it will wait for the previous write to finish.

If `writeUpdate` is called multiple times while a previous call hasn't finish, it will only write the most recent update.

`writeUpdateNow` is only the case where there is no previous call. Returning it would only sometimes mean "this update has finished writing".

You _could_ rewrite this function to return a Promise accurately tracking when the update has finished writing, but it would take a lot more code and work, and I don't see a pressing need to support that feature, relative to the amount of work it would require.

Reply:
I mean, none of the problem is API design – the ideal API seems like common sense to me. If it's as easy as you say, feel free to pull request it – I can't think of any reason I'd reject it.